File: solutions/2026-04-30/solution.py
# Container With Most Water
# Difficulty: Hard
# URL: https://leetcode.com/problems/container-with-most-water/

def maxSubarrays(nums): 
  n = len(nums)
  max_count = 0
  left = 0
  curr_sum = 0
  for right in range(n):
    curr_sum += nums[right]
    while curr_sum < 0 and left <= right:
      curr_sum -= nums[left]
      left += 1
    max_count = max(max_count, right - left + 1)
  return max_count

File: solutions/2026-04-30/solution.java
// Container With Most Water
// Difficulty: Hard
// URL: https://leetcode.com/problems/container-with-most-water/

public int maxSubarrays(int[] nums) {
  int n = nums.length;
  int maxCount = 0;
  int left = 0;
  int currSum = 0;
  for (int right = 0; right < n; right++) {
    currSum += nums[right];
    while (currSum < 0 && left <= right) {
      currSum -= nums[left];
      left++;
    }
    maxCount = Math.max(maxCount, right - left + 1);
  }
  return maxCount;
}